src/util/format_expr.cpp

```diff
@@ -128,10 +129,18 @@ static std::ostream &format_rec(
     else
       return os << src.pretty();
   }
-  else if(type == ID_unsignedbv || type == ID_signedbv)
+  else if(type == ID_unsignedbv ||
+          type == ID_signedbv ||
+          type == ID_c_bool)
     return os << *numeric_cast<mp_integer>(src);
   else if(type == ID_integer)
     return os << src.get_value();
+  else if(type == ID_string)
+  {
+    os << '"';
+    jsont::escape_string(id2string(src.get_value()), os);
```

Comment:
This should probably go into a separate header then, eg util/string_utils.h?

Reply:
Not obvious, since the escaping rules are language-specific. The benefit of calling this jsont::escape_string and keeping it in json.h is that it makes it clear that this uses JSON's rules for escaping.

Reply:
It seems weird to do json-specific escaping in a utility that isn't in any way specific to json?!

Reply:
The choice appears arbitrary.

Reply:
Now going for what .pretty() is printing now, i.e., escape() in lisp_expr.h